### DISCUSSION TITLE
How can I bypass the elastalert-create-index when running in docker?

### DISCUSSION THREAD

**Discussion by GraFindlay** [jertel/elastalert2#739] [Q&A] [answered]
Hi,
We are not allowed to write directly to the ES cluster and so need to bypass the create index step.

Is there an option to do this by passing an argument when the docker container runs?  If not, is there another way to do this?

Thanks,

**Comment by ferozsalam** [comment, marked as answer]
Are you using Kubernetes?

On Kubernetes, I have done this by setting a [custom](https://kubernetes.io/docs/tasks/inject-data-application/define-command-argument-container/) `command:` for the container that runs ElastAlert directly rather than calling the `run.sh` script that the Docker container runs by default.

I'm certain that Docker Compose will support something similar, although I'm unsure off the top of my head what the syntax would be.

**Reply by GraFindlay** [reply]
Thanks.  We plan to use ElastAlert2 on Kubernetes in the future but right now we are just trying to run it in a docker container so we can test out its function...so we really need some way of bypassing the create index step when running just in a docker container.

**Reply by jertel** [reply]
You can use the `--entrypoint` docker arg:

```
docker run -v <specify binds as documented> --entrypoint=elastalert jertel/elastalert2:2.3.0 --config /opt/elastalert/config.yaml <any optional args>
```

**Reply by GraFindlay** [reply]
@jertel That works perfectly, thanks
